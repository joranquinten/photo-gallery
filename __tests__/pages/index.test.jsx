import React from 'react';
import { shallow } from 'enzyme';

import Gallery from '../../pages/index';

describe('Gallery', () => {
  const mockContent = [
    {
      id: 'contentId',
      title: 'contentTitle',
    },
  ];

  it('should render without crashing', () => {
    expect(shallow(<Gallery type="type" content={mockContent} />)).toMatchSnapshot();
  });

  it('should render with propTypes provided', () => {
    expect(shallow(<Gallery type="type" slug="slug" content={mockContent} />)).toMatchSnapshot();
    expect(
      shallow(<Gallery type="type" pictureId="pictureId" content={mockContent} />)
    ).toMatchSnapshot();
    expect(
      shallow(
        <Gallery
          type="type"
          content={[
            ...mockContent,
            {
              id: 'contentId2',
              title: 'contentTitle2',
              slug: 'contentSlug2',
              description: 'contentDescription2',
              thumbnailUrl: 'contentThumbnailUrl2',
              tags: 'contentTags2',
              meta: 'contentMeta2',
            },
          ]}
        />
      )
    ).toMatchSnapshot();
  });
});
