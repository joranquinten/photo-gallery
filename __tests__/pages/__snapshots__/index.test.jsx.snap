// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery should render with propTypes provided 1`] = `
<Gallery
  content={
    Array [
      Object {
        "id": "contentId",
        "title": "contentTitle",
      },
    ]
  }
  pictureId={null}
  slug="slug"
  type="type"
/>
`;

exports[`Gallery should render with propTypes provided 2`] = `
<Gallery
  content={
    Array [
      Object {
        "id": "contentId",
        "title": "contentTitle",
      },
    ]
  }
  pictureId="pictureId"
  slug={null}
  type="type"
/>
`;

exports[`Gallery should render with propTypes provided 3`] = `
<Gallery
  content={
    Array [
      Object {
        "id": "contentId",
        "title": "contentTitle",
      },
      Object {
        "description": "contentDescription2",
        "id": "contentId2",
        "meta": "contentMeta2",
        "slug": "contentSlug2",
        "tags": "contentTags2",
        "thumbnailUrl": "contentThumbnailUrl2",
        "title": "contentTitle2",
      },
    ]
  }
  pictureId={null}
  slug={null}
  type="type"
/>
`;

exports[`Gallery should render without crashing 1`] = `
<Gallery
  content={
    Array [
      Object {
        "id": "contentId",
        "title": "contentTitle",
      },
    ]
  }
  pictureId={null}
  slug={null}
  type="type"
/>
`;
